### PR TITLE
Add SQuADShifts dataset

### PR DESCRIPTION
**Pull request by millerjohnp**
This PR adds the four new variants of the SQuAD dataset used in [The Effect of Natural Distribution Shift on Question Answering Models](https://arxiv.org/abs/2004.14444) to facilitate evaluating model robustness to distribution shift.

### CODE REVIEW

File: datasets/squadshifts/dummy/amazon/1.0.0/dummy_data/amazon_reviews_v1.0.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (lhoestq) [inline datasets/squadshifts/dummy/amazon/1.0.0/dummy_data/amazon_reviews_v1.0.json:1]: I think you can remove this file

File: datasets/squadshifts/dummy/new_wiki/1.0.0/dummy_data/new_wiki_v1.0.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (lhoestq) [inline datasets/squadshifts/dummy/new_wiki/1.0.0/dummy_data/new_wiki_v1.0.json:1]: same kere

File: datasets/squadshifts/dummy/nyt/1.0.0/dummy_data/nyt_v1.0.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (lhoestq) [inline datasets/squadshifts/dummy/nyt/1.0.0/dummy_data/nyt_v1.0.json:1]: same

File: datasets/squadshifts/dummy/reddit/1.0.0/dummy_data/reddit_v1.0.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (lhoestq) [inline datasets/squadshifts/dummy/reddit/1.0.0/dummy_data/reddit_v1.0.json:1]: same

File: datasets/squadshifts/dummy/amazon/1.0.0/dummy_data/amazon_reviews_v1.0.json (hunk 1 above)
Comment (millerjohnp) [reply datasets/squadshifts/dummy/amazon/1.0.0/dummy_data/amazon_reviews_v1.0.json]: Removed!

File: datasets/squadshifts/dummy/new_wiki/1.0.0/dummy_data/new_wiki_v1.0.json (hunk 2 above)
Comment (millerjohnp) [reply datasets/squadshifts/dummy/new_wiki/1.0.0/dummy_data/new_wiki_v1.0.json]: Removed

File: datasets/squadshifts/dummy/nyt/1.0.0/dummy_data/nyt_v1.0.json (hunk 3 above)
Comment (millerjohnp) [reply datasets/squadshifts/dummy/nyt/1.0.0/dummy_data/nyt_v1.0.json]: Removed

File: datasets/squadshifts/dummy/reddit/1.0.0/dummy_data/reddit_v1.0.json (hunk 4 above)
Comment (millerjohnp) [reply datasets/squadshifts/dummy/reddit/1.0.0/dummy_data/reddit_v1.0.json]: Removed

**Comment by lhoestq** [comment]
Very cool to have this dataset, thank you for adding it :)

**Pull request merged**